incident_viewer_demo/src/IspSelect/IspSelect.jsx

```diff
@@ -0,0 +1,135 @@
+
+import React, { PureComponent, PropTypes } from 'react';
+import Select from 'react-select';
+
+import Icon from '../Icon.jsx';
+
+import { colorsFor } from '../chart_support/color';
+
+import './react-select.scss';
+import './IspSelect.scss';
+import { cpus } from 'os';
+
+/**
+ * ISP Selection and display component
+ */
+export default class IspSelect extends PureComponent {
+
+  static propTypes = {
+    isps: PropTypes.array,
+    onChange: React.PropTypes.func,
+    selected: PropTypes.array,
+    placeholder: PropTypes.string,
+  }
+
+  static defaultProps = {
+    isps: [],
+    selected: [],
+  }
+
+  /**
+   * Main constructor. Bind handlers here.
+   */
+  constructor(props) {
+    super(props);
+    this.onAdd = this.onAdd.bind(this);
+    this.onRemove = this.onRemove.bind(this);
+  }
+
+  /**
+   * Callback to add a specified option to the
+   * selected ISP list
+   * @param {Object} {value: label:} option object to Add
+   */
+  onAdd(addValue) {
+    const { selected, onChange } = this.props;
+    // convert to options first
+    const values = this.getOptions(selected);
+    // addValue is a option Object
+    values.push(addValue);
+    if (onChange) {
+      onChange(values.map(value => value.value));
+    }
+  }
+
+  /**
+   * Callback to remove a specified value from the
+   * selected ISP list
+   * @param {Object} ISP object to remove
+   */
+  onRemove(removeValue) {
+    const { selected, onChange } = this.props;
+    const filtered = selected.filter((isp) => isp.client_asn_number !== removeValue.client_asn_number);
+    const values = this.getOptions(filtered);
+    if (onChange) {
+      onChange(values.map(value => value.value));
+    }
+  }
+
+  /**
+   * convert array of ISPs to an array of options to display
```

Comment:
Nit: capital "C" to start the docstring

Reply:
This component comes from the existing MLab client codebase.